packages/services/api/src/modules/auth/lib/authz.ts

```diff
@@ -0,0 +1,186 @@
+import { FastifyReply, FastifyRequest } from '@hive/service-common';
+import { HiveError } from '../../../shared/errors';
+import { isUUID } from '../../../shared/is-uuid';
+
+export type AuthorizationPolicyStatement = {
+  effect: 'allow' | 'deny';
+  action: string | string[];
+  resource: string | string[];
```

Comment:
I think we should make this a string enum. This way we at least in our code we get type safety for all possible resources today. If we introduce new resources we update one place